Azure : Split build steps and only run docs for release or on linux

themissingcow opened the pull request:
Draft PR as I don't have an arnold login for `installDependencies.sh`

johnhaddon commented on config/azure/build.yaml:
```diff
@@ -80,24 +87,35 @@ jobs:
        echo BUILD_TYPE=${{ parameters.buildType }}
        g++ --version
-       scons -j ${{ parameters.threads }} package ENV_VARS_TO_IMPORT=PATH BUILD_TYPE=${{ parameters.buildType }} DELIGHT_ROOT=$DELIGHT ARNOLD_ROOT=$ARNOLD_ROOT BUILD_DIR=./build INSTALL_DIR=./install/$(Gaffer.Build.Name) BUILD_CACHEDIR=sconsCache
+       rm -rf .sconsign.dblite .sconf_temp
```
Do we need this or is it left over from testing? I wouldn't expect either of these files to exist at this point.

themissingcow replied:
Totally, removed [here](https://github.com/GafferHQ/gaffer/pull/3877/commits/f6654a6b2b7706c799452e19b162104bc21729f5)

johnhaddon commented on config/azure/build.yaml:
```diff
@@ -55,10 +55,17 @@ jobs:
 
+   # Docs builds regularly hang on mac, temporarily disable them for non-release build
+   - script: |
+       echo "##vso[task.setvariable variable=Gaffer.Sphinx]/dev/null"
```
Can you explain why we are setting to `/dev/null`?

themissingcow replied:
An empty var doesn't work, but have updated the string [here](https://github.com/GafferHQ/gaffer/commit/1f55c9b14fa7911e47b9c394dbceb2b252bb4f37)

johnhaddon commented on config/azure/build.yaml:
```diff
@@ -80,24 +87,35 @@ jobs:
        ARNOLD_LICENSE_ORDER: none # Don't waste time looking for a license that doesn't exist
 
+   - script: |
+       echo BUILD_TYPE=${{ parameters.buildType }}
+       g++ --version
```
Do we need to repeat this? We shouldn't even be invoking the compiler in this build step, so it seems a little unnecessary.

themissingcow replied:
removed

themissingcow commented:
Thanks @johnhaddon. Sorry my brain seems to have been non-functional with this one. I've implemented all changes and squashed as requested.  I'll also update #3884 to use the same technique for disabling docs.
